Add configs for ucs-exporter

mithunmgore opened the pull request:
Added config files for alerts, templates.

Signed-off-by: Mithun Gore <mithun.gore@sap.com>

viennaa commented on prometheus-exporters/ucs-exporter/Chart.yaml:
```diff
@@ -0,0 +1,7 @@
+description: Exporter for UCSM/UCSC Server
+maintainers:
+  - name: Tommy Sauer (D066607)
+  - name: Mithun Gore (I355987)
```
please remove user ids

mithunmgore replied:
Done

viennaa commented on prometheus-exporters/ucs-exporter/Chart.yaml:
```diff
@@ -0,0 +1,7 @@
+apiVersion: v1
+name: ucs-exporter
+version: 0.1.0
```
let's start with 0.0.1

mithunmgore replied:
Done

viennaa commented on prometheus-exporters/ucs-exporter/alerts/service.alerts:
```diff
@@ -0,0 +1,14 @@
+  - alert: UcsServerPortLicenseExpired
+    expr: >
+      ucs_server_license{port_name=~".+"} == 0
+    for: 5m
```
15m please

mithunmgore replied:
Done

viennaa commented on prometheus-exporters/ucs-exporter/alerts/service.alerts:
```diff
@@ -0,0 +1,14 @@
+      meta: "ucs server {{ $labels.server }} port {{ $labels.port_name }} license is expired."
+    annotations:
+      description: "ucs server {{ $labels.server }} port {{ $labels.port_name }} license is expired."
+      summary: "ucs server {{ $labels.server }} port {{ $labels.port_name }} license is expired."
```
Can you add a hint what needs to be done or is not working? I don't  see a playbook linked yet. Something like:
`license is expired, server ran out of support` (I am making something up here)

mithunmgore replied:
I checked with Kapil and got to know that, It doesnt affect the vpod traffic, but to be compliant we need to add the new license. I have edited the message, pls check.

viennaa commented on prometheus-exporters/ucs-exporter/templates/ucs-exporter-deployment.yaml:
```diff
@@ -0,0 +1,65 @@
+             - -u
+             - {{ required ".Values.ucs.user missing" .Values.ucs.user }}
+             - -p
+             - {{ required ".Values.ucs.master_password missing" .Values.ucs.master_password }}
```
please change to `.Values.global.master_password`

mithunmgore replied:
Done

viennaa commented on prometheus-exporters/ucs-exporter/values.yaml:
```diff
@@ -0,0 +1,21 @@
+      prometheus: DEFINED-IN-REGION-SECRETS
+   user: DEFINED-IN-REGION-SECRETS
+   master_password: DEFINED-IN-REGION-SECRETS
+   inventory: "inventory"
```
what is this?

mithunmgore replied:
In program we need to get the inventory file name with -i flag, so I provided it here, not sure if its correct way. pls suggest.

viennaa replied:
This need to be the path then, where the inventory file sits. Please contact me to get this sorted.

viennaa commented on prometheus-exporters/ucs-exporter/values.yaml:
```diff
@@ -0,0 +1,21 @@
+   debug: "0"
+   limits:
+      memory: 100Mi
+      cpu: 450m
```
lets start with 250m here

mithunmgore replied:
Done

viennaa commented on prometheus-exporters/ucs-exporter/alerts/service.alerts:
```diff
@@ -0,0 +1,14 @@
+      meta: "ucs server {{ $labels.server }} port {{ $labels.port_name }} license is expired."
+    annotations:
+      description: "ucs server {{ $labels.server }} port {{ $labels.port_name }} license is expired."
+      summary: "ucs server {{ $labels.server }} port {{ $labels.port_name }} license is expired. Prod traffic won't be affected but to be compliant, apply new license."
```
What is prod traffic? Do you mean vpod traffic?

viennaa commented on prometheus-exporters/ucs-exporter/templates/ucs-exporter-deployment.yaml:
```diff
@@ -0,0 +1,65 @@
+           - name: DEBUG
+             value: {{ .Values.ucs.debug | quote }}
+           - name: INVENTORY
+             value: ucs-inventory
```
map the values.yaml inventory var here

viennaa commented:
@mithunmgore no updates?

mithunmgore commented:
@viennaa Sorry for late update, was working on netbox to get inventory.

Here in last commit, I have removed the "inventory" var/file and added config file to get all the inventory.
Config file will be present in the top most directory so added path as only 'config'. please check.
 I will create PR for ucs-exporter with config changes soon.  Thanks

viennaa commented on prometheus-exporters/ucs-exporter/values.yaml:
```diff
@@ -0,0 +1,21 @@
+      prometheus: DEFINED-IN-REGION-SECRETS
+   user: DEFINED-IN-REGION-SECRETS
+   master_password: DEFINED-IN-REGION-SECRETS
+   config: config
```
still the same problem as before, config does not map to anything here. it needs to be an absolute path in the container like /config/config.yaml

How are we feeding this in? Where is the data coming from?

viennaa replied:
Or is this coming from netbox now?

mithunmgore replied:
I have added the path to config. its "/ucs-exporter/config". Currently we are creating this config file manually, and I have added the "ucs_hostname_format" var which will get filled at run time with Netbox.

viennaa commented:
Please also add a service file to be queried from prometheus

viennaa commented on prometheus-exporters/ucs-exporter/templates/ucs-config.yaml:
```diff
@@ -0,0 +1,14 @@
+data:
+  config.yaml: |
+    netbox:
+        netbox_url: "https://netbox.global.cloud.sap"
+        netbox_token: "97bc508ab27df514c963435ae7097672efdaab2a"
+        ucs_hostname_format: "ucscluster-{}.cc.{}.cloud.sap"
+        regions: ["qa-de-1"]
```
this needs to be changed to `$values.ucs.netbox_url` etc
make sure in `values.yaml` you set it to `DEFINED-IN-REGION-SECRETS` and then add it to the region secrets.

mithunmgore replied:
I have updated, please check if the syntax is right.

viennaa reviewed (changes requested):
please change region mapping and values.yaml as stated

viennaa commented on prometheus-exporters/ucs-exporter/values.yaml:
```diff
@@ -0,0 +1,26 @@
+      enabled: false
+      prometheus: DEFINED-IN-REGION-SECRETS
+   user: DEFINED-IN-REGION-SECRETS
+   master_password: DEFINED-IN-REGION-SECRETS
```
since your mapping is to .Values.global.master_password, this is never used and can go.

viennaa reviewed (approved):
let's give it a go :).
